internal/command/deploy/machines.go

```diff
@@ -495,6 +504,46 @@ func (md *machineDeployment) configureLaunchInputForReleaseCommand(launchInput *
 	if _, present := launchInput.Config.Env["RELEASE_COMMAND"]; !present {
 		launchInput.Config.Env["RELEASE_COMMAND"] = "1"
 	}
+	if launchInput.Config.Guest == nil {
+		launchInput.Config.Guest = &api.MachineGuest{}
+	}
+	// The pointer dereferences around the MachineGuest logic are awkward, but it's done this way
+	// to ensure that the struct gets copied - that way, if something touches the relcmd machine,
+	// it doesn't affect the machine it stole its config from
+	desiredGuest := *defaultReleaseMachineGuest()
+	if !md.machineSet.IsEmpty() {
+
+		// Given a process group, return the MachineGuest with the most ram
+		maxRamInGroup := func(group string) *api.MachineGuest {
+			var (
+				maxRam = 0
+				guest  *api.MachineGuest
+			)
+			for _, lm := range md.machineSet.GetMachines() {
+				mach := lm.Machine()
+				if mach.Config != nil && mach.Config.Env != nil && mach.ProcessGroup() == group && mach.Config.Guest != nil {
+					if mach.Config.Guest.MemoryMB > maxRam {
+						guest = mach.Config.Guest
+					}
+				}
+			}
+			return guest
+		}
+
+		if maxRamInMachs := maxRamInGroup("app"); maxRamInMachs != nil {
+			desiredGuest = *maxRamInMachs
+		} else {
+			// No "main" app, sort lexicographically and just pick the first group
```

Comment:
could simplify if uses https://github.com/superfly/flyctl/blob/f92741d971911c1f542265c5e9dd696611e6e2c1/internal/appconfig/processconfig.go#L93

Reply:
Oh awesome, didn't know this existed. that'll make it a lot less ugly, thanks!